core/src/main/java/org/opensearch/remote/metadata/client/AbstractSdkClient.java

```diff
@@ -168,30 +168,36 @@ protected GetDataObjectResponse getGlobalResourceDataFromCache(GetDataObjectRequ
     }
 
     private GetDataObjectResponse replaceGlobalTenantId(GetDataObjectRequest request, GetDataObjectResponse response) {
-        response.source().put(TENANT_ID_FIELD_KEY, request.tenantId());
         GetResponse getResponse = response.getResponse();
         if (getResponse == null) {
             throw new OpenSearchStatusException(
                 "Cached response is null, please check configuration with system admin!",
                 RestStatus.INTERNAL_SERVER_ERROR
             );
         }
-        try {
-            JsonNode jsonNode = OBJECT_MAPPER.readTree(getResponse.toString());
-            ((ObjectNode) jsonNode.get("_source")).put(TENANT_ID_FIELD_KEY, Optional.ofNullable(request.tenantId()).orElse(DEFAULT_TENANT));
-            XContentParser parser = JsonXContent.jsonXContent.createParser(
-                NamedXContentRegistry.EMPTY,
-                LoggingDeprecationHandler.INSTANCE,
-                OBJECT_MAPPER.writeValueAsString(jsonNode)
-            );
-            return GetDataObjectResponse.builder().id(request.id()).parser(parser).source(response.source()).build();
-        } catch (IOException e) {
-            throw new OpenSearchStatusException(
-                "Failed to parse cached global response, please check configuration with system admin!",
-                RestStatus.INTERNAL_SERVER_ERROR,
-                e
-            );
+        if (response.getResponse().isExists()) {
+            response.source().put(TENANT_ID_FIELD_KEY, request.tenantId());
+            try {
+                JsonNode jsonNode = OBJECT_MAPPER.readTree(getResponse.toString());
+                ((ObjectNode) jsonNode.get("_source")).put(
+                    TENANT_ID_FIELD_KEY,
+                    Optional.ofNullable(request.tenantId()).orElse(DEFAULT_TENANT)
+                );
+                XContentParser parser = JsonXContent.jsonXContent.createParser(
+                    NamedXContentRegistry.EMPTY,
+                    LoggingDeprecationHandler.INSTANCE,
+                    OBJECT_MAPPER.writeValueAsString(jsonNode)
+                );
+                return GetDataObjectResponse.builder().id(request.id()).parser(parser).source(response.source()).build();
+            } catch (IOException e) {
+                throw new OpenSearchStatusException(
+                    "Failed to parse cached global response, please check configuration with system admin!",
+                    RestStatus.INTERNAL_SERVER_ERROR,
+                    e
+                );
+            }
         }
+        return response;
```

Comment:
(Style/readability) This is n't obviously an "else" condition. Better to put this as a quick return at line 178 when it's false, like:
```java
if (!getResponse.isExists()) {
    return response;
}
```
This keeps the logic together and you don't need to indent the section below it (which will make the diff easier to identify for review!)

Reply:
Done